### DISCUSSION TITLE
小狼毫输入法难用！每天打千字打一整年也没有改善！常用短语缺失。打到最后一个字母的时候突然把前面改错，诱骗用户污染自己词库。选一个“系”字竟然要按60多下键！

### DISCUSSION THREAD

**Discussion by WingFoxie** [rime/home#1727] [General]
我用小狼毫这么久了，总觉得很别扭。终于有一天我开始总结输入时的各种别扭。
结果短短11天就总结出这么多东西来：

小狼毫输入法竟然没有“痉挛”这个词？
没有“起效”
两个字念won𥥈𥦷（来跟我一起念：“汪汪！”），一个字念din𨈖
有“挣脱”没有“挣不脱”。
没有“戴到”只有“带到”
没有“刀斧”只有“棍棒”
没有“留得住”只有“留不住”
想打“起送价格”，结果输入“qisongjiage”出来的却是“**起送机阿哥**”
没有“弑神”
“探测不到”(tancebudao)。偏偏打到最后一个字母O的时候第一个字变成了“痰”
然后打“第一个字”打到“diyige”的时候还显示的是“第一个”，然后一按zi就变成“地一个字”了？？
类似的，“传送过来”打到最后的时候，突然变成了“穿送过来”
没有“副指挥官”
试图输入“把那些东西”，在打“banaxiedongxi”的时候，明明UI已经把音节正确分成了“ba na xie dong xi”，可是选字列表全都是读音为ban的字，然后要不断翻页，直到所有念ban的字（包括那一大堆极为生僻的字，甚至是“死字”）都过去之后，才能终于选上“把”字！！
“他们肯定会质问我”，打到前7个字的时候，还是正常的“他们肯定会质问”，然而打到最后一个字的时候突然就变成了“他们肯定会支问我”。把打到一半的“他们肯定会质问”按空格键上屏一次之后，再打“他们肯定会质问我”就正常了。
打“行啊”竟然出“醒啊”！
没有“酷炫”
没有“很费力”“嚼着”
打“jiaobulan”没有“嚼不烂”，可明明打“jiaobul”的时候是有“嚼不了”的。结果打出最后的“an”之后，连前面的“嚼不”也没有了，得三个字都选一遍！！
然后打这句“选一遍”时候，默认出来的是“选一边”。这个我是纠正了，可是剩下的呢？？“说一遍”，“做一遍”……所有的“动词+一遍”我都得一个个选？？？否则默认全是“动词+一边”。
同样的，“感叹道”“感慨道”“抱怨道”“调侃道”这样“动词+道”的也全都得自己选一遍，否则都是“动词+到”。
想打“更搞笑的是”，结果没有“更搞笑”，手动选一遍之后仍然无法直接打出“更搞笑的是”于是这个也得选一遍。
想打一句“我真是服了”，结果没有“我真是”，手动选的时候不小心把“我真时”也录入词库了！

没有“战端”“打晕”“自启”“删不了”
没有“四组”（但有“一组”“两组”“三组”）

没有“弯钩”。更气人的是，明明打“wangou”这六个字母的时候，已经正确把音节分成了“wan gou”，候选词是“玩够”，然而当我开始选字的时候，音节划分立马改为“wang ou"，于是我不但要选字，还要加个'手动分音节。
没有“拽倒”“此情此景”
想打“煮熟”，却只有“zhushu”能出这个词，打“zhushou”不行。手动选字的时候发现shou音中，“熟”字竟然排到了第二十位，在一堆罕见的字甚至是提手旁后面，差点就没找着！
然后没有“没找着”
想打“什么都知道”，结果默认出的是“什么都直到”。
“表演下去”，打到“biaoyanxia”的时候还是“表演下”，结果打最后一个字“去”的时候就突然变成了“表咽下去”？？

我明明打的是“zhenrongyi”（真容易），怎么唯一一个候选词是“真溶液”呢？？？多亏了这输入法让我知道“液”字还有yi的音！
这我就不明白了，我单独打“rongyi”的时候，候选词里没有“溶液”啊。怎么一加个“真”字就有了？

顺带一提装大词库并不能解决问题。比如就算没装大词库，词库里yishi共有77个候选词。
试想如果装个超级大词库包含所有的候选词，那么选起来也会相当麻烦的。
再举例来说，我的个人词库里其实一直只有“些什么”没有“写什么”。因为后者真的用不到。

然后快进到三个月之后，本以为我已经习惯了小狼毫输入法给我带来的各种恶心事了……
结果呢……
我想打：把绳子“系在”（jizai）树上，结果没有“系在”这个词，每页5个候选词的情况下，按了整整56下“=”键翻页，才终于找到了“系”（ji）字！！
全程我就跟个傻子一样！跟那些只会用眼从excel表格里找到需要的关键词的那些阿姨一样！每一页都得认真地看。
这些字的排序根本都是乱的！相同读音的情况下，不应该按笔划排序吗？？
**没有手动添加到词库的功能吗？**
没有办法**屏蔽掉那些生僻字**吗？那些字我真的永远都不用。恨死那个𨈖din字了！！这个字存在的唯一意义，就是让我输入带“定”字的词语时，不小心录入错误！
我现在需要的是做减法啊！能不能把那一大堆我这辈子都不会主动输入，却还要排在常见的“熟”字前面的那些破字屏蔽掉呢？？这些字对我来说唯一的用途就是干扰我输入。
说真的， 真的用那些字的人，会用全拼输入吗？？是不是如果我建议，**全拼模式直接屏蔽掉生僻字**，都没什么毛病呢？？

然后我是说56下。结果呢，当时没选，没录入词库。之后我再选一次的时候，足足按了60多下才再次找到“系”字！我估计是我为了打出这个词来，打了几遍“系”字，然后排序又变了？？亏我还一边按一边数着。
没有办法直接手动编辑词库吗？？我就是因为打不出这个词才需要录入，然而想录入词库必须得先排除万难，用输入法自带的别扭选字打出这个词来？？就不能有个工具，让我选中一个词，直接添加进词库， 设置好读音吗？？

我是想打开词库文件自己编辑的。不就是个txt文档么？可是我打开一看，每一列的含义我不知道，每个词库里的词后面都带几个数，不知道什么意思。
最气人的是，这文档前面加了好几行注释……**怎么就没有一行注释解释一下每一列的含义呢？？**

一个输入法做成这样我也真是服了。彻底服了。
然后最新版本直接UI都变模糊了！！
别让我帮你们试究竟是从哪个版本起变模糊的。我每更新一次版本电脑得重启至少两次，第一次重启之后，就连动个鼠标都是极其卡顿的，害得我以为要因为个输入法重装系统了！再重启一次才恢复正常。
并且刚卸载旧版本之后，尽管卸载全程什么窗口也没开，之后仍然点什么什么崩溃，只能在卡顿和惊慌中勉强点到“重启”。
从旧版本更新到UI模糊的新版本，再降级回旧版本。我就重启了不知道多少次电脑。
卸载新版本——重启——动鼠标都卡，于是再重启！——装回旧版本——重启——动鼠标都卡顿，于是再重启！——终于恢复正常。
降级回UI不模糊的旧版本，我就重启了4次！结果之后才意识到忘了截图，至少截个模糊的UI啊，唉……

补充下文字描述吧。我现在输入法用PS4主题，结果在我这边看到，只有第一个候选词前面的“1.”是模糊的，剩下的是清晰的。
结果更新到最新版本之后，全变模糊了！
我用的是大于100%的系统DPI，如果无法复现BUG，把DPI调高点儿再试试看。
没办法，本来一张截图能说清楚的事儿，可我现在实在不敢为了这件事装个新版本了。谁知道又要重启多少次！

还有那个该死的分音节功能。我知道在输入中可以按'号强制分音节。比如“xi'an”就是强制打“西安”而不是“先”。
可是有没有强制不分音节的功能啊？？比如那个“起送机阿哥”我要怎么办？？只能傻乎乎选字？我能不能把“起送价格”的“jia”音设为强制不分？能不能加入连续两个`号强制不分音节的功能？？如输入“qisongji''age”？？

诱骗用户打错字这点也是够可以的！
打到最后一个字的时候突然把前面某个字改错了。打到最后一个字声母的时候还都正确，打完韵母就又错了？
甚至唯独打到最后一个字母的时候错了？比如下面这个例子。
想打“今天干点儿什么呢？”打到“干点儿”的时候，偏偏输入到最后一个字母的时候，才突然把第一个字换成“敢”，诱骗我打错字！打到倒数第二个字母的时候，输入了“gandiane”的时候，还是“干点呃”呢！简直防不胜防了！

补充一下“动个鼠标都卡顿”的情况：
我在网上找了一圈，发现有个人遇到同样的卡顿。结果他找了个叫“LatencyMon”的工具检测了一下，结果发现是英伟达显卡驱动导致的卡顿。
**于是我也下载“LatencyMon”检测了一下，发现是一个叫"nvlddmkm.sys"的驱动导致卡顿，就是英伟达显卡驱动导致的。**
可我又能怎么办！驱动已经更新到最新版了！
而且卡顿的问题还不定时出现。
比如说，**我电脑连着开三天不重启，结果还能只有第二天出问题，第一天和第三天不出问题。**
我现在还得用旧版本小狼毫输入法，因此不会时刻卡顿。但是玩某个旧游戏的时候，是会出现卡顿的。电脑一直开着，结果今天就怎么都会卡，明天就又好了。

要是用小狼毫输入法最新版，问题就更严重了。每次点开开始菜单的时候会卡顿好一阵子。恢复后，快速连续打开不会卡顿。但等几秒钟，干点儿别的，再打开就又卡了。
似乎是打开开始菜单的时候，有一瞬间GPU频率会飙到最高，然后就出现了严重的卡顿。（正常情况下，打开开始菜单时，GPU频率也会会一瞬间飙高，但不会卡。）
还有就是打开任何需要管理员权限的应用时，弹出UAC授权框的时候，也会严重卡顿。系统音效也变成卡顿拉长的样子。

结果之后有一次系统更新重启后又卡顿了，我可没用最新版小狼毫输入法啊。再重启一次不卡了。
之后又有一次，系统更新重启后卡顿了，再重启，还卡。再重启，还卡。重启了六次还卡！
于是我直接把系统重装了一遍，先保守治疗，保留文档也保留应用。
重装后终于不卡了。
然后再重启一次又卡了！！！
然后再次装显卡驱动，选中“清洁安装”选项，装完后还是卡。（注：我想了想还是不要用非官方的DDU，即显卡驱动卸载工具来卸载驱动了。还是只用官方自带的选项吧。）
再重启一次终于不卡了，暂时不敢重启了。

然后现在还是玩某个旧游戏的时候会出现卡顿。而且电脑连续开着，都能某一天卡顿，第二天又完全不卡顿。至少现在只限于那一个游戏了。
游戏太老了，用帧率检测软件都检测不到掉帧。只显示一直60帧满帧，并且输入延迟也不受影响，似乎游戏的内部逻辑并没有受到卡顿影响，只是画面在丢帧。

总之：
**我现在每天都要打一千字到几千字，已经持续一年多了。**结果还是每天至少遇到好几次上述那些恶心的情况。
问题似乎永远也没完没了。因此我不敢想象那些只是偶尔用这输入法打字的人，要怎么才能受得了。
我天天养词库还搞成这样？！
就这句，又发现没有“养词库”。唉……连抱怨都不敢抱怨。

我把百科翻了个底朝天，发现上述问题有个别的其实有办法解决，比如按左箭头键指定音节（如“弯钩”的例子）。但绝大多数是没办法的！

这日子到底什么时候是个头儿啊……要到什么时候，才能真的把词库养成“只要输入明确对应一个输出的都不需要选字”啊？
本来想说“似乎是所有‘短语’性质的候选词都缺少的”，比如“行啊”。
可是没有“jingluan”这个……“jingluan”是个名词性质的候选词啊，不是短语性质的候选词啊。
说真的，我在文章一开始说这个问题了。同样在用小狼毫输入法的你，能顺利打出“jingluan”这个词吗？就是表示肌肉不自觉抽搐的那个“jingluan”？
体会一下要自己在一堆长得差不多的字里，选择是哪个jing哪个luan的感觉吧！

打出来了？好，接下来打“jizai”！系绳子的系！

**Comment by LEOYoon-Tsaw** [comment]
實事求是地說，拼音輸入體驗確實不如主流輸入法吧，優勢在於靈活和自定義，如果沒有這方面需求其實沒必要堅持用Rime

**Reply by puer** [reply]
我其实看中的是安全性。现在的输入法，都要同步你的词库到云端的。一不小心，你的密码，重要内容，隐私就泄露了。

**Reply by yivifu** [reply]
你可以试一下添加辅码反查。可以用部件/拼音输入方案来反查，也可以用五笔画来反查。雾凇拼音输入方案最新版默认用部件/拼音输入方案反查，不过需要记忆一下部件在键盘上的布局（比五笔字形容易记，因为基本上按照部件拼音的第一个字母布置的），改用五笔画则可以无须记忆辅码，用hspnz对应横竖撇捺折笔画（点当作捺）按顺序输入就行，这样定位词库中没有的词时比较容易找到候选字，一点也不比搜狗拼音差。修改辅码反查方案可参阅：[小狼毫输入法雾凇拼音输入方案辅码由默认的部件拆字/拼音输入方案修改为五笔画方案](https://blog.csdn.net/yivifu/article/details/148336749)

**Comment by lotem** [comment]
息怒。

@fxliang 小狼毫有沒有集成 octagram？octagram 應當成爲標配~ 組詞的問題加語言模型應該能好很多。

**Reply by fxliang** [reply]
原来就没有吧，再加上我也不用拼音方案，所以我没有动相关的东西

**Reply by lotem** [reply]
沒事。你專心看前端。留意一下卡頓的問題。

**Comment by lotem** [comment]
這個帖子記錄了很多使用中發現的問題。您費心了。
但是這些問題不容易一下全部解決。

維護輸入法的前端程序、算法、詞庫等各個部分，由衆多開發者按照各自的技能和興趣分擔。
用家的貢獻也是很重要的。
更有效的辦法是把每一個獨立的問題記錄成一筆[工單](https://github.com/rime/home/issues)。方便大家分工協作、各個擊破。

**Comment by smaxad47** [comment]
你说的3/4的问题都可以通过使用换输入方案来解决,比如雾凇,白霜.
兼容性问题,我自己一般都保留微软的英语输入状态来打游戏之类的.

**Comment by isudfv** [comment]
我试了下「痉挛」和「系鞋带」也打不出来，赶紧换成了雾凇，你提到的大部分词都能正常输入了。

**Comment by WingFoxie** [comment]
我想试一下雾凇拼音的，结果没试成还亏大了……
下载WeaselSchemaSetupV2.1.exe安装，再重新部署，不行。选雾凇拼音时只能打英文。
于是我又装了一遍，这次选了清空配置文件的选项。还是只能打英文。
过了一段时间我才意识到，**用户词库的“备份”也是放在“用户文件夹”里的**。尽管是在一个单独建的sync文件夹里，但WeaselSchemaSetupV2.1.exe清空配置文件时，试图把“用户文件夹”整个清空，也就把那个sync文件夹也删了。
我就这样删除掉了自己养了一年的词库，连备份都给删了。
也就是说当初我**不但要“输出词典快照”，还要把生成的“luna_pinyin.userdb.txt”转移别处保存才行。**否则，用小狼毫输入法自带功能导出的备份，会在清空用户配置文件时被删掉！
并且还不能事先选择导出文件的路径，必须首先也导出到那个会被清掉的sync文件夹里，然后手动复制到别处。

并且那个“雾凇拼音”我怎么也没法部署到能用的状态。
直接从雾凇拼音那里下载个“full.zip”解压到用户文件夹里再部署也不行，这次连皮肤显示都不正常了。
又用“WeaselSchemaSetupV2.1.exe”装了一遍，皮肤正常了但是雾凇拼音方案还是只能打英文。
尽可能把用户文件夹手动清空了（有几个文件提示占用中不让删，那我要怎么清空？？），再把那个“full.zip”解压到用户文件夹里，还是不行。
并且有时候必须连续部署两次，第一次部署的时候，UI会跳过选主题的环节，然后会给我重置到默认主题。于是为了换主题又得部署第二遍。
结果就是部署了不下10遍，还是没用上“雾凇拼音”。

打开一个叫“rime什么什么log.ERROR什么什么”的文件，搜索里面的“attempt to index a nil value (field 'map')”和“attempt to concatenate a nil value (field 'search_key_alt')”错误，也没什么头绪。
然后还留了个后遗症。每次按“Win+空格”切出小狼毫输入法，使用那个自带的“不知道这个字发什么音Wiki里也没写——月拼音的时候”，都要按Shift+F1，然后按2，然后按4，从繁体输入切成简体输入。（编辑：后来发现是我选错输入方案了。“朙月拼音”默认繁体，另一个叫朙月拼音“简体字”啊不“簡化字”啊不“简化字”的，才是默认简体。“朙月拼音简化字”。还有那个字念ming，ming月。）
这……
然后“雾凇”这个名字起得也不好，因为有一个两点水的“雾凇”还有一个三点水的“雾淞”，还都在小狼毫输入法自带词库里（我用户词库都清空了，100%可确定是自带的）。一不留神搜了三点水的，还疑惑怎么突然间什么都搜不出来了……

**Reply by lotem** [reply]
最好用默認的配置。

**Comment by smaxad47** [comment]
换吧 Rime很需要动手能力的,WIN下体验本身就不如Mac甚至不如Linux, 至少能有理解和修改YAML的能力
试试卡饭? 完全离线注重隐私(曾经好像还被曝光用了rime的代码),但是一键安装,有好看的GUI,可以尝试一下

**Comment by WingFoxie** [comment]
唉，我就直接问吧……**雾凇拼音和白霜拼音是不兼容0.14.3版本么？怎么也没人说一句呢**？所有相关项目的说明文件里都没有提。
现在情况是，我折腾了好久，最终**似乎成功把雾凇的词库嫁接到了0.14.3自带的朙月拼音上面**。
同时**尽可能屏蔽了朙月拼音原本的词库**。

第一个问题自然就是：“真的嫁接上了吗？”
因为反正我都把词库给清了，我用原版朙月拼音打“他们肯定会质问我”的时候，发现出的是“他们肯定会指纹我”。
而现在配置成这样，直接出“他们肯定会质问我”。

然后我**把主帖里抱怨的所有东西都又试了一遍。问题减少了很多，只说仍然有问题的部分**：

还是没有“戴到”。
“起送价格”出的是“奇送价格”，不过选个字就好了。但这明明也属于“只有一种可能”的情况吧？
默认还是没有“选一遍”，不过别的“动词+一遍”似乎都有。（我在把“选一遍”正式录入词库之前试的！所以肯定是真的有。）
类似的情况，默认还是没有“感叹道”，但剩下几个“动词+道”，“感慨道”“抱怨道”“调侃道”是有的。
还是没有“自启”。
没有“拽倒”。

可以说是好很多了。
剩下要么都能直接出，要么不是“只有一种可能”，不在候选词头一个情有可原。比如“弯钩”排在了“玩够”后面。

然后我主帖里说默认词库里yishi共有77个候选词。
然而现在配置成这样，yishi的候选词只剩30个了。
现在默认词库里既有“些什么”也有“写什么”。
“把绳子系在树上”的“系在”jizai自然也有了，就是第一次打这句话得选一遍字。
**不再受到wonwon和dindin的骚扰了。不再有那些不能用系统默认字体显示出来的字了。**

**我是这么嫁接的：**

注意：**要大改“用户文件夹”里的“luna_pinyin.dict.yaml”文件，把里面原有的内容几乎全都删了**！所以建议先备份一下这个文件，到桌面上或者别的什么地方。

在rime_frost.dict.yaml还是rime_ice.dict.yaml文件里，都有都有下面一些内容。这几行似乎就是负责调用那些词库文件的。

```
import_tables:
  - cn_dicts/8105     # 字表
  # - cn_dicts/41448  # 大字表（按需启用）
  - cn_dicts/base     # 基础词库
  - cn_dicts/ext      # 扩展词库
  # - cn_dicts/tencent  # 腾讯词向量（大词库，部署时间较长）
  - cn_dicts/others   # 一些杂项
...
```

注意我引用的部分最后有一行三个点儿么？利用这找到嫁接的位置。
原luna_pinyin.dict.yaml文件里哪个一行三个点儿的行，后面还有太多太多的东西。全都删掉！
然后从“import_tables:”起把rime_ice.dict.yaml里那几行都复制过来，就放在原文件的“use_preset_vocabulary: true”那行下面吧。
三个点儿后面还有一堆说是什么“按需启用”的，我也复制过来了，或许不复制也行。
原luna_pinyin.dict.yaml文件里，那行use_preset_vocabulary: true我也在前面加上了#号，相当于删掉了？

说白了就是一边调用新装的几个，“cn_dicts”词库文件夹里的文件，一边尽可能屏蔽掉原版朙月拼音的词库。那个现在已经是负资产了。

**那文件luna_pinyin.dict.yaml最终改成了这样**：
（带<>尖括号的部分是我的描述，不是实际文件内容。）
```

<文件开始>
<前面一堆带#号的行。我就不展示了。>
---
name: luna_pinyin
version: "2020.08.03"
sort: by_weight
#use_preset_vocabulary: true
import_tables:
  - cn_dicts/8105     # 字表
  # - cn_dicts/41448  # 大字表（按需启用）
  - cn_dicts/base     # 基础词库
  - cn_dicts/ext      # 扩展词库
  - cn_dicts/tencent  # 腾讯词向量（大词库，部署时间较长）
  - cn_dicts/others   # 一些杂项

  # 建议把扩展词库放到下面，有重复词条时，最上面的权重生效
  # - cn_dicts/mydict
...

# 按需启用：

# 大写字母
# 可在拼音方案下直接输入大写字母（Shift + 字母），大写字母参与造句。
<后面也一堆行我也不展示了>
<文件结束>
```

这文件**改之前800多KB，改完后只剩4KB**了。
然后**再“部署”，方案只选择“朙月拼音·简化字”，就可以部署这个嫁接词库后的拼音方案了**。（顺便一提中间那个“·”号可以用Shift+8然后再按3打出来。）
然后再输入，就是我上面测试出来的效果。

或许效果比起正式装上雾凇还是打折扣的？因为我试着打雾凇项目页面里举的那若干个长句例子，发现**只有大约一半的例子是和截图里一样直接正确的**。
但肯定是比原版朙月拼音好太多了。

**Reply by yrljroli** [reply]
查阅雾凇拼音的使用说明、其中明确的指出：

> 要使用雾凇拼音默认提供的所有功能，请保证您的 RIME 前端提供的 librime 版本 ≥ 1.8.5 且含有 librime-lua 依赖。

https://github.com/iDvel/rime-ice/blob/main/README.md

而librime是在0.15.0更新至1.8.5的。你认为

> 所有相关项目的说明文件里都没有提。

显然不确。

**Comment by markwu** [comment]
@lotem 不好意思，攔截一下話題。我看到你有說有 `octagram`可以用，請問現在小狼毫的 librime 是否已經包含了 octagram，是否只要將 grammar.yaml 複製到 Rime 中即可啟用 `octagram`？或是要做其他的設置？

**Reply by lotem** [reply]
rime.dll 包含了所需插件，但是没有打包语言模型的数据文件，请参考 https://github.com/lotem/rime-octagram-data 的说明添加。

**Reply by markwu** [reply]
謝謝你，瞭解了！已經添加。

**Reply by markwu** [reply]
@lotem 測試了一下，實在說不出有沒有比較好。

例如，原本我可以順利打出 `添加後`，可是使用了 `octagram` 後，反而出現了 `天家後`，因為在八股文中有出現 `家後` 這個詞，雖然 `添加` 的詞頻較高。

我的測試是把所有 `build` 和 `*.userdb` 刪除後重新部屬測試。

**Reply by lotem** [reply]
octagram 不是词典，记的是前后两个词的搭配。

**Reply by markwu** [reply]
了解。所以出現這樣的結果，代表在 octagram 中判斷 `天 + 家後`  比 `添加 + 後`  有更高的機率？還是權重嗎？

**Reply by lotem** [reply]
無法重現啊。你列舉的結果很不合理。
如果排除用戶詞典影響。可能用錯了模型，要用 `hant` 分支的[zh-hant-t-essay-bgw.gram](https://github.com/lotem/rime-octagram-data/blob/hant/zh-hant-t-essay-bgw.gram)。也可能方案部署不成功。也可能某個版本程序有BUG。

**Reply by markwu** [reply]
了解，我先關掉我自己的輸入法，用預設的注音輸入法來測試看看再回報。感謝幫忙。

**Reply by markwu** [reply]
@lotem 太有趣了，我找出原因了，可是不知道為什麼。我轉換了新酷音和小麥輸入法的預設詞庫，裡面包含了單字和多字的詞組，所以認為應該就不需要 `terra_pinyin.dict.yaml`，因此就把他移除了。

使用上一切也都很正常。

可是當我加入 `octagram` 測試時就發生這個事情了。就是 `天 + 家後` 和 `添加 + 後` 的差別。

我剛剛用 Rime 的注音輸入法測試，發現一切正常。後來我嘗試移除所有我自己添加的詞庫，然後用回  `terra_pin.dict.yaml` 就正常了。

等一下，再來慢慢把詞庫添加回去，看看是哪個詞庫讓他不正常。

**Reply by markwu** [reply]
全部詞庫添加回去後也都正常，代表的是 `terra_pinyin.dict.yaml` 裡面的詞頻設定讓 `octagram` 變正常。

**Reply by markwu** [reply]
有意思，大概發現不正常的詞庫，來自於小麥注音的預設詞庫，不確定原因是什麼。

**Reply by markwu** [reply]
問題找到了，小麥注音的詞頻轉換後大小差異太大，所以`後`的詞頻非常非常的大。可能是這個問題，讓 `octagram` 被影響。

這就回到我上次問的問題，好幾個都有詞頻的詞庫，我該要如何配置。

**Comment by yrljroli** [comment]
事实上、<u>rime</u>是一个输入法**引擎**、他可以用来搭载适合的输入方案。他的优势本在于高度可自定义、不在于词库。

自带的<u>朙月拼音</u>对于普通用户而言、也的确不如主流输入法好用。拼音用户可以考虑导入<u>搜狗输入法</u>的词库。

就你提出的几点。

> 没有手动添加到词库的功能吗？

编缉词库文件。

> 没有办法屏蔽掉那些生僻字吗？

换用无生僻字的词库。

> 我是想打开词库文件自己编辑的。不就是个txt文档么？可是我打开一看，每一列的含义我不知道，每个词库里的词后面都带几个数，不知道什么意思。

词库一般是yaml、而非txt。虽然二者都是纯文本。

我查阅了<u>朙月拼音</u>的词库、在70832行的文件中、止有2300个词后带有数字、且止有一个百分数。《每个词库里的词后面都带几个数》的现像不确。

词库的格式一般是这样：

> 龍韜虎略	long tao hu lve

显然、前为输出字符、后为输入码。我不认为这须要专门设置注释、况且这些内容皆有教程。详参[Rime_description.md](https://github.com/LEOYoon-Tsaw/Rime_collections/blob/master/Rime_description.md)。

**Reply by WingFoxie** [reply]
我想要手动修改的词库是“用户词库”，不是输入方案自带的。

就是用户文件夹里，luna_pinyin.userdb子文件夹里面的.ldb文件。
这文件显然不好直接编辑；分为好几个文件，打开是乱码，并且文件好像直接被输入法进程锁定了？

但可以**用“【小狼毫】用户词典管理”里面的“导出词典快照”功能，让用户文件夹里的sync文件夹里，出现导出后的luna_pinyin.userdb.txt文件。**
似乎此时手动编辑导出的文件后再导入，就可以追加用户词库内容。

然后我这luna_pinyin.userdb.txt文件打开前几行大概是这个样子：

```
# Rime user dictionary
#@/db_name	luna_pinyin.userdb
#@/db_type	userdb
#@/rime_version	1.5.3
#@/tick	64
#@/user_id	xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
a 	啊	c=0 d=0.1 t=43
an zhuang 	安裝	c=1 d=1 t=6
ba 	把	c=1 d=1.00032 t=63
```

当然后面还有很多很多行，就不全放出来了。我都把词库不慎全清了，再养词库到现在，也有6000多条了。
后面有的条目，c值最大的能达到63，d值最大的有3.7左右，t值最大则有上万的。

**你知道每一行的c d t都是什么意思吗？**
这文件开头整整六行以#开头的注释，却没一行解释下这三个参数的含义。项目文档里似乎也没有？

而且，无论是输入方案自带词库还是用户词库，里面内容似乎是**按照字母顺序排列**的。
**加入新项目的时候是一定要手动加在正确的顺位吗？（万一加错了位置……）还是可以直接加在末尾？**
这点项目文档里似乎也没明确说明。

**Reply by lotem** [reply]
用戶詞典記錄的是真實的輸入習慣。詞典快照反應的是這些真實的輸入數據，無法手動修改。導入的快照文件也應當是先前從用戶詞典導出的文件。

擴充固態詞典內容，首推在詞典文件的YAML配置中使用import_tables從其他詞典源文件導入碼表的方法。
一定要導入到用戶詞典的話，應該用「導入文本碼表」。

**Reply by lotem** [reply]
參考文檔
https://github.com/rime/home/wiki/UserGuide#%E7%94%A8%E6%88%B6%E8%A9%9E%E5%85%B8%E7%AE%A1%E7%90%86

沒有說明的就是不推薦修改~

**Reply by sidottu** [reply]
用五筆沒你說的那些問題，另外用戶添加詞庫在rime第三方極點方案那邊有個工具能添加（自己搞個擴展詞庫），不過由於我是五筆自動造詞很方便所以沒用過他那個，像“系在”,可以秒造好（單字優勢），所以用不上他這工具
![image](https://github.com/user-attachments/assets/174fff6c-b7bc-4494-bcab-8017b5b56e92)


**另外生僻字五筆的方案可以關掉，常用和擴展就是生僻字開關，拼音一些方案也應該能做到**

![image](https://github.com/user-attachments/assets/fe88a6d1-ccaa-4c4d-9c2e-f9ce2c354a1f)
`極點的添加工具，因爲自動造詞效率更高，且重碼越少越好所以我用不上`
![image](https://github.com/user-attachments/assets/eab9bb72-db6a-4361-aa9d-e358b3924dec)

**Comment by ajtcmkg** [comment]
您说的以上问题我初见rime也是头疼,不过我并不懂编程,所以仅能提供些使用方面的经验,希望您能少走弯路
1.学会分词,比如您上面提到的例子"把那些东西",我安装了一个全新干净的Rime,初次键入"ba na xie dong xi"候选词为:"把那写东西",你可以按下Tab来进行分词:(1)键入ba na xie dong xi (2)按一下Tab将输入框的光标移动至"把" (3)通过空格或数字键来选"把"(4)其后再依次选择"那些"和"东西",或者直接选择"那些东西".  重点是通过词典中已有的"分词"来组合出"新词",比如"那些""东西"两词很明显是常见的词汇
    分词结构: 把|那些|东西
    错误分词结构: 把那|些东|西
2.载入外部词典,这一节我不展开讲了,因为有很多优秀教程,不过你要注意要将外部词典作为"固态词典"导入
3.使用形码,五笔郑码之类,选字准确,rime和搜狗输入法差别不大,由于rime可定制性,定制后要更胜一筹
4.使用他人提供的现成方案,比如github上有雾凇拼音,四叶草拼音
5.弃用rime,或许在这里的评论区说有些不正确,但这确实是一个可行的选择,而且个人在现实中向多人推荐过rime后失败率比较高,其实许多人连带有图形化界面的搜狗输入法都不愿意去调整设置,如果你注重隐私,讨厌搜狗之流的广告,Windows和macOS自带的中文输入法也是不错的选择

**Comment by lotem** [comment]
難用還用，是不是自虐啊。
既然想清楚了自己想要什麼，就自己做。
